src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -256,6 +249,54 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         _AttachedHandlers(*this, nullptr);
     }
 
+    TerminalConnection::ITerminalConnection ControlCore::Connection()
+    {
+        return _connection;
+    }
+
+    // Method Description:
+    // - Setup our event handlers for this connection. If we've currently got a
+    //   connection, then this'll revoke the existing connection's handlers.
+    void ControlCore::Connection(const TerminalConnection::ITerminalConnection& newConnection)
+    {
+        if (_connection)
+        {
+            _connectionOutputEventRevoker.revoke();
+        }
+
+        // Subscribe to the connection's disconnected event and call our connection closed handlers.
+        _connectionStateChangedRevoker = newConnection.StateChanged(winrt::auto_revoke, [this](auto&& /*s*/, auto&& /*v*/) {

```

Comment:
If you replace connection A with connection B, should we emit a "connection state changed" notification to the TermControl consumer?

Reply:
That seems sensible, though, won't the `connection.Start` typically do that?

Reply:
I am an unhinged maniac that sets `Connection` to one that is already fully started and running! I am not necessarily calling `.Start`

Reply:
1. you're an unhinged maniac
2. you're the ConnectionState czar - what do you think? Is it valid for someone to get a notification that they're in the current state?

Reply:
I'm thinking that maybe we should dispatch a notification _iff_ the state is different between old and new connection.